src/test/java/seedu/address/logic/parser/getparsers/GetMedicationCommandParserTest.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.logic.parser.getparsers;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.getcommands.GetMedicationCommand;
+import seedu.address.model.person.MedicationContainsKeywordsPredicate;
+
+public class GetMedicationCommandParserTest {
+
+    private GetMedicationCommandParser parser = new GetMedicationCommandParser();
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        // empty input
+        assertParseFailure(parser, "     ",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, GetMedicationCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_garbledArg_returnsGetMedicationCommand() {
+        // extra prefixes
+        String userInput = "/inp /fn /outp";
+        GetMedicationCommand expectedGetMedicationCommand = prepareCommand(userInput);
+        assertParseSuccess(parser, userInput, expectedGetMedicationCommand);
+
+        // garbled input
+        userInput = "asdfghjkl";
+        expectedGetMedicationCommand = prepareCommand(userInput);
+        assertParseSuccess(parser, userInput, expectedGetMedicationCommand);
+    }
+
+    @Test
+    public void parse_validArgs_returnsGetMedicationCommand() {
+        // no leading and trailing whitespaces
+        GetMedicationCommand expectedGetMedicationCommand = prepareCommand("paracetamol ibuprofen");
+        assertParseSuccess(parser, "paracetamol ibuprofen", expectedGetMedicationCommand);
```

Comment:
I believe this exact test input has been used in other classes too. Maybe we should look at abstracting this away into its own variable too.